test/journeys/lib/widgetPages/space/messaging.js

```diff
@@ -0,0 +1,360 @@
+import path from 'path';
+
+import {assert} from 'chai';
+
+import MainSpaceWidget from './main';
+
+const uploadDir = path.resolve(process.cwd(), 'test/journeys/lib/helpers/assets');
+
+export default class MessageWidgetPage extends MainSpaceWidget {
+  get hasTextArea() { return this.browser.isVisible(this.elements.textArea); }
+
+  get hasActivityList() { return this.browser.isVisible(this.elements.activityList); }
+
+  get hasModalWindow() { return this.browser.isVisible(this.elements.modalWindow); }
+
+  get hasModalDeleteButton() { return this.browser.isVisible(this.elements.modalDeleteButton); }
+
+  get hasFlagButton() {
+    const {elements} = this;
+    return this.browser.isVisible(`${elements.lastSuccessfulActivity}${elements.flagButton}`);
+  }
+
+  get lastSuccessfulActivityText() { return this.browser.getText(this.elements.lastSuccessfulActivityText); }
+
+  get lastFileSize() { return this.browser.getText(this.elements.fileSize); }
+
+  constructor(props) {
+    super(props);
+
+    const activityContainers = '//div[contains(@class, "ciscospark-activity-item-container")]';
```

Comment:
Why are some of these defined up here and others within the elements object?

Reply:
If it's used to compose a selector in the `elements` object, we have to define it first before we can use it. I tried to keep anything that isn't used in a test or getter out of the actual `elements` object